[Wang Xinrong] iP

wang-xinrong opened the pull request:
### Duke
> "We all love pulling all-nighters!" - Robort Toroonto

Benefits of using Duke include:
- text-based
- simplistic GUI
- **fun**

You can:
1. download it from [here](https://github.com/wang-xinrong/ip/releases/tag/A-Jar)

_It is **freeeee**_

Features:
- [x] Managing tasks
- [ ] Do tasks for you :satisfied:

it is developed using Java, take at look at this `main` function
`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

wang-xinrong committed:
added Liv and HorizontalLine classes, updated README.md with planning records

wang-xinrong committed:
added echo functionality

wang-xinrong committed:
implemented add and list

wang-xinrong committed:
implemented mark and unmark

wang-xinrong committed:
implemented ToDo, Deadline and Event classes

wang-xinrong committed:
implemented aumotated TextUI Testing

wang-xinrong committed:
implemented basic error handling

wang-xinrong committed:
implemented task deletion

wang-xinrong committed:
Add save functionality

wang-xinrong committed:
Add load functionality

wang-xinrong committed:
Merge branch 'branch-level-7'

wang-xinrong committed:
add Dates and Times functionality

wang-xinrong committed:
Merge branch 'branch-level-8'

wang-xinrong committed:
add Ui.class

wang-xinrong committed:
add Parser.class

wang-xinrong committed:
add TaskList.class

wang-xinrong committed:
add Storage.class

wang-xinrong committed:
clean up code

wang-xinrong committed:
Merge branch 'branch-a-moreoop'

wang-xinrong committed:
add packages

wang-xinrong committed:
Merge branch 'branch-a-packages'

wang-xinrong committed:
resolve README.md conflict

wang-xinrong committed:
add JUnit and add test classes

wang-xinrong committed:
add test for Task class

wang-xinrong committed:
add jar build process

wang-xinrong committed:
sort members by access modifier

wang-xinrong committed:
add find functionality

wang-xinrong committed:
add some javadoc comments

wang-xinrong committed:
add javadoc comments

wang-xinrong committed:
Merge branch 'branch-level-9'

wang-xinrong committed:
resolve conflict

wang-xinrong committed:
Add JavaFX tutorial

wang-xinrong committed:
add JavaFX Tutorial 2

wang-xinrong committed:
add iteration 1 of JavaFX tutorial

wang-xinrong committed:
add GUI in progress

wang-xinrong committed:
add functional GUI

wang-xinrong committed:
add checkstyle

wang-xinrong committed:
Update GUI

wang-xinrong committed:
implement GUI

wang-xinrong committed:
Merge branch 'branch-level-10'

wang-xinrong committed:
resolve conflicts

jeong-jaeho commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public String getResponse(String input) {
+        try {
+            return parser.ProcessInputReturnString(input);
```
Shouldn't the parser's method be in camelCase?

jeong-jaeho commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,213 @@
+     * @param localDateTime DateTime to be converted.
+     * @return String representing dateTime in command format.
+     */
+    public static String convertDateTimeToCommandFormat(LocalDateTime localDateTime) {
```
I like how you use name to explain what the variable is!

jeong-jaeho commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,178 @@
+public abstract class Task {
+    private boolean isDone;
+    protected TaskType type;
+    public static String delimiter;
```
I like how you try to abide by the rule of using nouns for things and verbs for actions.

jeong-jaeho commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,178 @@
+    public static String command;
+    protected static String dataStringSplitter = " \\| ";
+    // temporary measure before storage related methods are migrated over to the storage class
+    protected static String storageDataStringSplitter = " | ";
```
I like how you try to avoid 'too short' names to explain what the variable does. It's not excessively lengthy, so it only helps me understand your code better.

jeong-jaeho commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,118 @@
+public class TaskList {
+    private static LinkedList<Task> tasks = null;
+    private static LinkedList<Task> lastFilteredTasks = null;
+    private static TaskList instance = null;
```
Perhaps would it be better to name it more specific to the purpose it serves?

jeong-jaeho commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,144 @@
+package duke;
```
Overall, I like how you use appropriate language and avoid slangs or names that are not meaningful within contexts/times.

jeong-jaeho reviewed (commented):
LGTM. Just some nits to fix.

wang-xinrong committed:
Add varargs

wang-xinrong committed:
update JAR main file setting

ChillinRage commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,144 @@
+    /**
+     * Toggles active state of Duke.
+     */
+    public void ToggleActiveState() {
```
Is it better to use camelCasing for "toggleActiveState" to maintain naming consistency?

ChillinRage commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,213 @@
+        throw new CommandNotFoundException(input);
+    }
+
+    public String ProcessInputReturnString(String input) throws InputException {
```
Is it better to use switch-case instead of if blocks?

ChillinRage commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,101 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+import duke.parser.Parser;
+import duke.task.Task;
+import duke.task.TaskList;
```
I like that the import packages are well separated and is in a sorted order.

ChillinRage commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,57 @@
+    public void setUpTask(String input) throws MissingInputFieldException {
+        try {
+            input = input.trim();
+            if (!input.contains(command)) throw new RuntimeException("not todo");
```
Would it be better to follow the coding standard for an if block?

ChillinRage commented on src/main/java/duke/ui/HorizontalLine.java:
I like that a horizontal line is also broken down into a class itself.

ChillinRage reviewed (commented):
Looks good!

wang-xinrong committed:
activate assertions through gradle setting

wang-xinrong committed:
add assert and refactor code

wang-xinrong committed:
Merge pull request #2 from wang-xinrong/branch-a-assertions

add assert and refactor code

wang-xinrong committed:
refactor code till Task.class

wang-xinrong committed:
refactor code

wang-xinrong committed:
Merge branch 'branch-a-codequality'

wang-xinrong committed:
add gitHub actions

wang-xinrong committed:
implement comparison between tasks

wang-xinrong committed:
implement removeDuplicates and sort

wang-xinrong committed:
implement undo

wang-xinrong committed:
Merge branch 'branch-extensions'

wang-xinrong committed:
Currently there is no available user guide for Liv.

This undermines the usability of the product. It is also failing iP requirement.

Let's update the README.md as the user guide.

wang-xinrong committed:
update README.md

wang-xinrong committed:
update README.md again

wang-xinrong committed:
prepare for submission

wang-xinrong committed:
prepare for submission

wang-xinrong committed:
Merge branch 'master' of https://github.com/wang-xinrong/ip

wang-xinrong committed:
Update exit command

wang-xinrong committed:
Update GUI

wang-xinrong committed:
Update GUI screenshot

wang-xinrong committed:
Handle wrong date&time input errors

wang-xinrong committed:
Merge branch 'branch-improvements'

wang-xinrong committed:
Remove unnecessary print statement

wang-xinrong committed:
Update command names

wang-xinrong committed:
Update UG
